internal/ui/backup/progress.go

```diff
@@ -48,16 +44,24 @@ type Summary struct {
 	archiver.ItemStats
 }
 
+const (
+	stateStarted = 1 << iota
+	stateFinished
+)
+
 // Progress reports progress for the `backup` command.
 type Progress struct {
+	mu sync.Mutex
+
 	interval time.Duration
 	start    time.Time
 
-	totalCh     chan Counter
-	processedCh chan Counter
-	errCh       chan struct{}
-	workerCh    chan fileWorkerMessage
-	closed      chan struct{}
+	state            uint8
```

Comment:
It would be simpler to just use `started` and `finished` flags. It took me a while to notice that `state` is a bitfield.

Reply:
Replaced by boolean fields scanStarted and scanFinished.